src/importer.cpp

```diff
@@ -767,11 +789,29 @@ ComponentPtr flattenComponent(const ComponentEntityPtr &parent, ComponentPtr &co
         applyEquivalenceMapToModel(rebasedMap, model);
 
         // Copy over units used in imported component to this model.
+        auto clonedImportModel = importModel->clone();
         StringStringMap unitsNamesToReplace;
         for (const auto &u : requiredUnits) {
-            const std::string originalName = u->name();
-            const std::string newName = addUnitsAvoidingNameClash(model, u);
+            // If the required units are imported units, we will resolve those units here.
+            size_t unitsIndex = 0;
+            UnitsPtr flattenedUnits = nullptr;
+            while (unitsIndex < clonedImportModel->unitsCount()) {
+                auto foundUnits = clonedImportModel->units(u->name());
+                if (foundUnits == nullptr) {
+                    unitsIndex += 1;
+                } else {
+                    flattenUnitsTree(clonedImportModel, u, unitsIndex);
+                    flattenedUnits = clonedImportModel->units(unitsIndex);
+                    break;
+                }
+            }
+
+            auto replacementUnits = (flattenedUnits != nullptr) ? flattenedUnits : u;
+
+            const std::string originalName = replacementUnits->name();
+            const std::string newName = addUnitsAvoidingNameClash(model, replacementUnits);
```

Comment:
Would remove the above empty line and replace `const std::string` with `auto`.

Reply:
I have the empty line so that the above statement is isolated and clear.